src/main/java/com/aws/greengrass/authorization/WildcardTrie.java

```diff
@@ -15,70 +14,28 @@
- * - isWildcard: If current Node is a valid MQTT wildcard (#, +)
- * - matchAll: if current node should match everything. Could be MQTTWildcard or a wildcard and will always be a
- *   terminal Node.
+ * - isWildcard: If current Node is glob wildcard (*)
  */
 public class WildcardTrie {
     protected static final String GLOB_WILDCARD = "*";
-    protected static final String MQTT_MULTILEVEL_WILDCARD = "#";
-    protected static final String MQTT_SINGLELEVEL_WILDCARD = "+";
-    protected static final String MQTT_LEVEL_SEPARATOR = "/";
     protected static final char wildcardChar = GLOB_WILDCARD.charAt(0);
-    protected static final char multiLevelWildcardChar = MQTT_MULTILEVEL_WILDCARD.charAt(0);
-    protected static final char singleLevelWildcardChar = MQTT_SINGLELEVEL_WILDCARD.charAt(0);
-    protected static final char levelSeparatorChar = MQTT_LEVEL_SEPARATOR.charAt(0);
 
     private boolean isTerminal;
-    private boolean isTerminalLevel;
     private boolean isWildcard;
-    private boolean isMQTTWildcard;
-    private boolean matchAll;
     private final Map<String, WildcardTrie> children =
             new DefaultConcurrentHashMap<>(WildcardTrie::new);
 
     /**
      * Add allowed resources for a particular operation.
-     * - A new node is created for every occurrence of a wildcard (*, #, +).
-     * - Only nodes with valid usage of wildcards are marked with isWildcard or isMQTTWildcard.
+     * - A new node is created for every occurrence of a wildcard *
      * - Any other characters are grouped together to form a node.
-     * - Just a '*' or '#' creates a Node setting matchAll to true and would match all resources
      *
      * @param subject resource pattern
      */
     public void add(String subject) {
         if (subject == null) {
             return;
         }
-        if (subject.equals(GLOB_WILDCARD)) {
-            WildcardTrie initial = this.children.get(GLOB_WILDCARD);
-            initial.matchAll = true;
-            initial.isTerminal = true;
-            initial.isWildcard = true;
-            return;
-        }
-        if (subject.equals(MQTT_MULTILEVEL_WILDCARD)) {
-            WildcardTrie initial = this.children.get(MQTT_MULTILEVEL_WILDCARD);
-            initial.matchAll = true;
-            initial.isTerminal = true;
-            initial.isMQTTWildcard = true;
-            return;
-        }
-        if (subject.equals(MQTT_SINGLELEVEL_WILDCARD)) {
-            WildcardTrie initial = this.children.get(MQTT_SINGLELEVEL_WILDCARD);
-            initial.isTerminal = true;
-            initial.isMQTTWildcard = true;
-            return;
-        }
-        if (subject.startsWith(MQTT_SINGLELEVEL_WILDCARD + MQTT_LEVEL_SEPARATOR)) {
-            WildcardTrie initial = this.children.get(MQTT_SINGLELEVEL_WILDCARD);
-            initial.isMQTTWildcard = true;
-            initial.add(subject.substring(1), true);
-        }
-
         add(subject, true);
```

Comment:
how is the first entry a terminal?

Reply:
It starts with assuming terminal, unless it encounters a `*` where in it adds a node with all the previous characters and mark it terminal false

Reply:
Is it ever false, can the parameter be removed?

Reply:
Yes, it is needed and would be false.